### PR TITLE
[Term entry] Python Built-in-Function: .Repr()

### PR DESCRIPTION
**Pull request by Mboatella25**
### Description

- Added the .repr() function for Python.
- Closes Issue #2152 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by SSwiniarski** [comment]
@Mboatella25 yo seem to have committed this with an e-mail that isn't linked to your GitHub account. Please add the "Marie" email to your GitHub account and sign the [CLA](https://cla-assistant.io/Codecademy/docs?pullRequest=2183).

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Methods'
+  - 'Functions'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:13]:
```suggestion
---

```
Add a line break after metadata.

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+The **`repr()`** function returns a printable string describing the object. It gives us the string representation passed implicitly to the eval() function.
+A class can control what this function returns for its instances when called by defining a [`__repr__()`](https://www.codecademy.com/resources/docs/python/dunder-methods/repr) dunder method.
+
+## Syntax
+
+The `repr()` function accepts a single object as parameter. 
+
+
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:21]:
```suggestion
```
Remove extra line break

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+```pseudo
+repr(object)
+```
+
+
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:26]:
```suggestion
```
Remove extra line break.

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+## Codebyte Example
+
+Here, two instances of `repr()` are called.
+- The first one returns the default value (a string with the name and address of the object, enclosed in angle brackets).
+- The second defines the output string and implements the `__repr__()`method.
+
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:32]:
```suggestion
```
Remove extra line break.

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    pass
+  def __repr__(self):
+    return "Undefined variable"
+var = Variable()
+
+print(repr(cte))
+
+print(repr(var))
+
+
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:49]:
```suggestion
```
Remove extra line break

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+Here, two instances of `repr()` are called.
+- The first one returns the default value (a string with the name and address of the object, enclosed in angle brackets).
+- The second defines the output string and implements the `__repr__()`method.
+
+
+```codebyte/python
+cte = "constant"
+
+
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:37]:
```suggestion
```
Remove extra line break

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+class Variable:
+  def __init__(self):
+    pass
+  def __repr__(self):
+    return "Undefined variable"
+var = Variable()
```
Comment (SSwiniarski) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:43]:
```suggestion

var = Variable()
```
Add line break.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@Mboatella25  I did a first review and left some comments, all about line breaks in your entry.

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
+---
+Title: '.repr()'
```
Comment (yangc95) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:2]:
```suggestion
Title: 'repr()'
```

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+Description: 'Returns a printable string describing the object.'
```
Comment (yangc95) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:3]:
```suggestion
Description: 'Returns a printable string describing the object that is passed in.'
```

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+Tags:
+  - 'Methods'
+  - 'Functions'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+The **`repr()`** function returns a printable string describing the object. It gives us the string representation passed implicitly to the eval() function.
+A class can control what this function returns for its instances when called by defining a [`__repr__()`](https://www.codecademy.com/resources/docs/python/dunder-methods/repr) dunder method.
```
Comment (yangc95) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:16]: Either add a new line between these two or combine onto one line.

Also:

```suggestion
The **`repr()`** function returns a printable string describing the object that is passed in. It gives us the string representation passed implicitly to the `eval()` function.

A class can control what this function returns for its instances when called by defining a [`__repr__()`](https://www.codecademy.com/resources/docs/python/dunder-methods/repr) dunder method.
```

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+The `repr()` function accepts a single object as parameter. 
+
+```pseudo
+repr(object)
+```
+
+## Codebyte Example
+
+Here, two instances of `repr()` are called.
+- The first one returns the default value (a string with the name and address of the object, enclosed in angle brackets).
```
Comment (yangc95) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:29]:
```suggestion
Here, two instances of `repr()` are called:

- The first one returns the default value (a string with the name and address of the object, enclosed in angle brackets).
```

File: content/python/concepts/built-in-functions/terms/repr/repr.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+    pass
+  def __repr__(self):
+    return "Undefined variable"
+    
+var = Variable()
+
+print(repr(cte))
+
+print(repr(var))
+
```
Comment (yangc95) [inline content/python/concepts/built-in-functions/terms/repr/repr.md:46]:
```suggestion
print(repr(cte))
print(repr(var))
```

**Review by yangc95 — COMMENTED** [review]
@Mboatella25 Conducted a second review. Please tag me once you'd like me to take another look

**Comment by Mboatella25** [comment]
@yangc95 Fixed the issues of the second review.

**Review by yangc95 — APPROVED** [review]

**Comment by yangc95** [comment]
Thanks for this entry @Mboatella25 🙌 Feel free to make additional edits to this entry at any time.

Screenshot 2023-06-02 at 8 00 33 PM

It looks as if you're anonymous. If you want to be credited follow the instructions below to add your GitHub username to your Codecademy profile. If not, there is nothing more you need to do.
To add your GitHub username to your Codecademy profile:
1.) Add your GitHub Username to your Codecademy Profile Details, you can get there on your desktop by pressing on your profile picture in the upper right-hand corner and selecting "Profile" in the drop-down box:
githubusername

Make sure to set your visibility to “everyone”:
Screenshot 2023-04-10 at 4 07 30 PM


2.) Then log out and log back into Codecademy using your GitHub login:
githublogin

If the steps above do not work, it may be due to an email or local configuration issue.
In addition, we may reach out in the future if any adjustments to these entries are required.

cc: @SSwiniarski

**Pull request merged**